api.py

```diff
@@ -0,0 +1,277 @@
+            '/spiders',
+            view_func=self.list_spiders,
+            methods=['GET'],
+        )
+        self.add_url_rule(
+            '/spiders',
+            view_func=self.run_spider,
+            methods=['POST'],
+        )
+        self.add_url_rule(
+            '/spiders/<int:spider_id>',
+            view_func=self.close_spider,
+            methods=['DELETE'],
+        )
+        self.add_url_rule(
+            '/database',
+            view_func=self.import_db,
+            methods=['POST'],
+        )
+        self.add_url_rule(
+            '/database',
+            view_func=self.export_db,
+            methods=['GET'],
+        )
+        self.add_url_rule(
+            '/database',
+            view_func=self.clear_scraps,
+            methods=['DELETE'],
+        )
+        self.add_url_rule(
+            '/crawls',
+            view_func=self.list_crawls,
+            methods=['GET'],
+        )
+        self.add_url_rule(
+            '/crawls',
+            view_func=self.stop,
+            methods=['DELETE'],
+        )
+        self.add_url_rule(
+            '/',
+            view_func=self.home,
+            methods=['GET'],
+        )
+
+    def home(self):
+        routes = [
+            {
+                "url": "/spiders",
+                "method": "GET"
+            },
+            {
+                "url": "/spiders",
+                "method": "POST",
+                "arguments": {
+                    "spider": "name of the spider to run"
+                }
+            },
+            {
+                "url": "/spiders/:spider_id",
+                "method": "DELETE",
+                "arguments": {
+                    "spider_id": "uuid of the spider to close"
+                }
+            },
+            {
+                "url": "/crawls",
+                "method": "GET"
+            },
+            {
+                "url": "/crawls",
+                "method": "DELETE"
+            },
+            {
+                "url": "/database",
+                "method": "GET"
+            },
+            {
+                "url": "/database",
+                "method": "POST",
+                "arguments": {
+                    "file": "json file containing the database dump"
+                }
+            },
+            {
+                "url": "/database",
+                "method": "DELETE"
+            }
+        ]
+        result = self._get_meta_response({"routes": routes})
+        return jsonify(result), 200
+
+    def list_spiders(self):
+        spiders = self.process.spider_loader.list()
+        return jsonify({"spiders": spiders, "status": "success"}), 200
+
+    def run_spider(self):
+        post_data = request.get_json()
+        spider = post_data.get('spider')
+        if spider == 'who_iris':
+            spider = who_iris_spider.WhoIrisSpider()
+        elif spider == 'nice':
+            spider = nice_spider.NiceSpider()
+        else:
+            return '', 400
```

Comment:
I think the right HTTP code is 404 for a resource that does not exist. https://stackoverflow.com/questions/1533510/rest-standard-for-get-on-a-resource-that-doesnt-exist. 

Not super important, up to you